src/main/java/io/github/aurora/scheduler/offers/HttpPluginConfig.java

```
@@ -0,0 +1,88 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.github.aurora.scheduler.offers;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+
+import com.google.gson.Gson;
+
+import org.apache.commons.io.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class HttpPluginConfig {
+  private static final Logger LOG = LoggerFactory.getLogger(HttpPluginConfig.class);
+  private String endpoint = "http://localhost:9090";
+  private Config config;
+  private static final int DEFAULT_LOG_STEP = 1000;
+  /*
+    aurora-plugin.json file:
+    {
+      "host": "localhost",
+      "port": 9090,
+      "debug": true,
+      "logStepInTaskNum": 100
+    }
+   */
+  public HttpPluginConfig() {
+    final String configFile = "/etc/aurora-scheduler/http-endpoint.json";
+    // load file
+    String jsonStr = null;
+    try {
+      jsonStr = FileUtils.readFileToString(new File(configFile), StandardCharsets.UTF_8);
+    } catch (IOException io) {
```

What are you doing with exceptions here after catching? If there is no config file how can the plugin run? You should throw an exception and let the caller know if you don't have a default config.

---

Two choices here:

1. Fail the scheduler.
2. Throw a warning message saying the plugin is not configured properly.

Aurora has traditionally erred on the side of failing the scheduler to warn users about misconfiguration.

---

we don't need config file any more.